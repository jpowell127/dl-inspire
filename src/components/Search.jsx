import React from "react";
import PropTypes from "prop-types";
import { Input, Button } from "semantic-ui-react";

export default function Search({
  getUsers,
  handleSearchTermChange,
  onKeyPress
}) {
  return (
    <Input
      inverted
      size="large"
      action={{
        color: "pink",
        icon: "search",
        onClick: getUsers
      }}
      placeholder="Search Behance"
      onChange={handleSearchTermChange}
      onKeyPress={onKeyPress}
    />
  );
}

Search.propTypes = {
  getUsers: PropTypes.func.isRequired,
  handleSearchTermChange: PropTypes.func.isRequired
};
