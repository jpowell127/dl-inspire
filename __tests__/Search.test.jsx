import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import Search from "../src/components/Search";

test("Search renders correctly", () => {
  const component = shallow(
    <Search getUsers={() => {}} handleSearchTermChange={() => {}} />
  );
  expect(component).toMatchSnapshot();
});
