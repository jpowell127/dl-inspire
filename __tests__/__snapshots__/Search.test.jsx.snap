// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getUsers={[Function]}
    handleSearchTermChange={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "action": Object {
        "color": "pink",
        "icon": "search",
        "onClick": [Function],
      },
      "inverted": true,
      "onChange": [Function],
      "placeholder": "Search Behance",
      "size": "large",
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "action": Object {
          "color": "pink",
          "icon": "search",
          "onClick": [Function],
        },
        "inverted": true,
        "onChange": [Function],
        "placeholder": "Search Behance",
        "size": "large",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
